ci/test-conformance-eks.sh

```diff
@@ -0,0 +1,228 @@
+    eksctl create cluster \
+      --name ${CLUSTER} --region ${REGION} --version=${K8S_VERSION} \
+      --nodegroup-name workers --node-type ${AWS_NODE_TYPE} --nodes 2 \
+      --ssh-access --ssh-public-key ${SSH_KEY_PATH} \
+      --managed
+    if [[ $? -ne 0 ]]; then
+        echo "=== Failed to deploy EKS cluster! ==="
+        exit 1
+    fi
+
+    mkdir -p ${WORKSPACE}/jenkins/out/
+    eksctl utils write-kubeconfig \
+      --cluster=${CLUSTER} --kubeconfig=${KUBECONFIG_PATH}/kubeconfig
+    export KUBECONFIG="${KUBECONFIG_PATH}/kubeconfig"
+
+    sleep 5
+    if [[ $(kubectl get svc) ]]; then
+        echo "=== EKS cluster setup succeeded ==="
+    else
+        echo "=== EKS kubectl is not configured correctly! ==="
+        exit 1
+    fi
+}
+
+function deliver_antrea_to_eks() {
+
+    echo "=== Configuring Antrea for cluster ==="
+
+    # EKS service CIDR is currently assigned by AWS,
+    # either from 172.20.0.0/16 or 10.100.0.0/16 depending on the range of VPC
+    # https://forums.aws.amazon.com/thread.jspa?messageID=859958
+    k8s_svc_addr=$(kubectl get svc -A | grep kubernetes | awk '{print $4}')
+
+    if [[ $k8s_svc_addr == 10.100.* ]]; then
+    	 k8s_svc_cidr='10.100.0.0/16'
+    elif [[ $k8s_svc_addr == 172.20.* ]]; then
+    	 k8s_svc_cidr='172.20.0.0/16'
+    else
+    	 echo "Cannot determine EKS serviceCIDR!"
+         exit 1
+    fi
+
+    set +e
+    worker_node_ip_1=$(kubectl get nodes -o wide | sed -n '2 p' | awk '{print $7}')
+    node_mtu=$(ssh -o StrictHostKeyChecking=no -l "ec2-user" $worker_node_ip_1 \
+      'export PATH=$PATH:/usr/sbin; ip a | grep -E eth0.*mtu | cut -d " " -f5')
+
+    sed -i "s|#defaultMTU: 1450|defaultMTU: ${node_mtu}|g" ../build/yamls/antrea-eks.yml
+    sed -i "s|#serviceCIDR: 10.96.0.0/12|serviceCIDR: ${k8s_svc_cidr}|g" ../build/yamls/antrea-eks.yml
+    echo "defaultMTU set as ${node_mtu}"
+    echo "seviceCIDR set as ${k8s_svc_cidr}"
+
+    kubectl apply -f ../build/yamls/antrea-eks.yml
+    set -e
+
+    kubectl rollout status --timeout=2m deployment.apps/antrea-controller -n kube-system
+    kubectl rollout status --timeout=2m daemonset/antrea-agent -n kube-system
+
+    echo "=== Antrea has been deployed for EKS cluster ${CLUSTER} ==="
+}
+
+function run_conformance() {
+    echo "=== Running Antrea Conformance and Network Policy Tests ==="
+
+    chmod +x run-k8s-e2e-tests.sh
+    ./run-k8s-e2e-tests.sh --e2e-conformance --e2e-network-policy > eks-test.log
+
+    if grep -Fxq "Failed tests:" eks-test.log
+    then
+        echo "Failed cases exist."
+        touch TEST_FAILURE
+    else
+        echo "All tests passed."
+    fi
+
+    echo "=== Cleanup Antrea Installation ==="
+    for antrea_yml in ../build/yamls/*.yml
+    do
+        kubectl delete -f ${antrea_yml} --ignore-not-found=true || true
+    done
+
+    if !(test -f TEST_FAILURE); then
+        echo "=== SUCCESS !!! ==="
+        exit 0
+    fi
+    echo "=== FAILURE !!! ==="
```

Comment:
just noticed something: if the tests fail, shouldn't we exit the function with code 1? With the current code, I'm not sure that the script is actually going to exit with a non-zero code. Could you confirm?

Reply:
You are right. I think essentially we should not exit 0 here because we still want the cleanup to happen if running this script in setup-cleanup mode.

Reply:
maybe you want to store the status code in a variable and exit with it at the end, after cleanup?

Reply:
We probably don't want cleanup to exit with 1 for failing CI tests since we will get a email notification everytime CI fails, instead of when cleanup fails?

Reply:
Think I found a workaround for this. Updated the script

Reply:
For `--cleanup-only`, you don't run the tests, so it should only fail if cleanup fails?

For `--setup-only` (respectively when both are run), I think you should exit with 1 if the CI tests fail (respectively when either the tests or the cleanup fail).

Reply:
> For `--cleanup-only`, you don't run the tests, so it should only fail if cleanup fails?
> 
> For `--setup-only` (respectively when both are run), I think you should exit with 1 if the CI tests fail (respectively when either the tests or the cleanup fail).

Yes essentially this is done in the latest patch.